waspc/examples/crud-testing/src/server/tasks.ts

```diff
@@ -0,0 +1,41 @@
+import { createCrudOverrides } from "@wasp/crud/tasks.js";
+
+export const overrides = createCrudOverrides({
```

Comment:
Ok, so what are our options for overrides?

One way is to provide Prisma clauses, like @infomiho did, which is quite interesting, as it allows user to still use a lot of stuff that we implemented and override just a part of it.
But, at this early stage where our CRUD logic is not super smart, that will likely not be enough.

So, we can go the opposite route and give them full control -> they override it all, by providing the function that is same like Operation: `(context, args)` ... .
That could work for sure. Question is, what is left then, not much? Should we just let them link to an actual Operation then? smth like `create: CreateTask`? They could just use that operation directly of course, but this way it is nice because they can use it as `tasks.create`.
I wonder now if we should also allow them to group some other operations here, that we haven't predicted? So they could be used also via `tasks.myoperation`, just for the elegance of it / discoverability? Hmmm.

OK, let's get back to track. So we can do declarative override that is not strong enough, we can do full override which feels a bit much. Can we do something in between? We could provide "hooks": they could provide logic that is done before the actual operation (they can do stuff like validation and auth here), logic that is done after the operation (they can sanitize data and similar here). I imagine that would be somewhere in between regarding the expresivity? Some more hooks? Maybe we could have this, and also have total override as the last option.

To summarize:
- I think we should offer complete override at the moment. I am not sure if by providng ExtImport directly, or by linking to existing Operation -> maybe linking to existing Operation.
- But I would also try to offer finer-grained hooks, that happen before and/or after the core IO. You will be able to better figure out if this makes sense or not and to what degree, but I imagine this could work ok for now. And then full override is used if this is not enough. But this would put some foundation work for finer customization.

Reply:
We are now doing complete overrides (impl are written like for operations)